jdbc-source/src/main/java/org/springframework/cloud/stream/module/jdbc/source/JdbcSourceConfiguration.java

```diff
@@ -0,0 +1,122 @@
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.cloud.stream.module.MaxMessagesProperties;
+import org.springframework.cloud.stream.module.PeriodicTriggerConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.integration.core.MessageSource;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.SourcePollingChannelAdapterSpec;
+import org.springframework.integration.dsl.channel.MessageChannels;
+import org.springframework.integration.dsl.core.Pollers;
+import org.springframework.integration.dsl.support.Consumer;
+import org.springframework.integration.jdbc.JdbcPollingChannelAdapter;
+import org.springframework.integration.scheduling.PollerMetadata;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.springframework.transaction.PlatformTransactionManager;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A module that reads data from an RDBMS using JDBC and creates a payload with the data.
+ *
+ * @author Thomas Risberg
+ */
+@EnableBinding(Source.class)
+@EnableConfigurationProperties({JdbcSourceProperties.class, MaxMessagesProperties.class})
+@Import({PeriodicTriggerConfiguration.class})
+public class JdbcSourceConfiguration {
+
+	@Autowired
+	private JdbcSourceProperties properties;
+
+	@Autowired
+	DataSource dataSource;
+
+	@Autowired
+	PlatformTransactionManager transactionManager;
+
+	@Autowired
+	@Bindings(JdbcSourceConfiguration.class)
+	Source source;
+
+	@Bean
+	TaskExecutor taskExecutor() {
+		return new ThreadPoolTaskExecutor();
+	}
+
+	@Bean
+	@ConditionalOnProperty(value = "split", havingValue = "true", matchIfMissing = true)
+	public IntegrationFlow splitFlow() {
+		return IntegrationFlows.from("splitter")
+				.split("payload")
+				.channel(MessageChannels.executor(this.taskExecutor()))
+				.channel(source.output())
+				.get();
+	}
+
+	@Bean
+	public MessageSource<Object> jdbcMessageSource() {
+		JdbcPollingChannelAdapter jdbcPollingChannelAdapter =
+				new JdbcPollingChannelAdapter(dataSource, properties.getQuery());
+		jdbcPollingChannelAdapter.setMaxRowsPerPoll(properties.getMaxRowsPerPoll());
+		if (properties.getUpdate() != null) {
+			jdbcPollingChannelAdapter.setUpdateSql(properties.getUpdate());
+		}
+		return jdbcPollingChannelAdapter;
+	}
+
+	@Bean
+	public PollerMetadata poller() {
+		return Pollers.fixedRate(properties.getFixedDelay(), TimeUnit.SECONDS).
```

Comment:
Yes, I would agree. Inherited from XD where we do use fixed-delay poller, although I think it should have been fixed-rate in the first place. Will change to fixedDelay to stay consistent and then switch to #146 configs when available.